Menus/ScenarioGUI.cpp

```diff
@@ -26,2022 +12,1051 @@
+			set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+			line(drawBitmap, lineStartX + 1, lineStartY, lineEndX + 1, lineEndY, drawColor);
+			line(drawBitmap, lineStartX - 1, lineStartY, lineEndX - 1, lineEndY, drawColor);
+			line(drawBitmap, lineStartX, lineStartY + 1, lineEndX, lineEndY + 1, drawColor);
+			line(drawBitmap, lineStartX, lineStartY - 1, lineEndX, lineEndY - 1, drawColor);
+		}
+
+		// Draw a circle around the selected site point.
+		Vector sitePos = m_PlanetCenter + m_SelectedScene->GetLocation() + m_SelectedScene->GetLocationOffset();
+		int sitePosX = sitePos.GetFloorIntX();
+		int sitePosY = sitePos.GetFloorIntY();
+		int circleRadius = 8;
+
+		blendAmount = 195 + RandomNum(0, 30);
+		set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+		circle(drawBitmap, sitePosX, sitePosY, circleRadius, drawColor);
+
+		blendAmount = 120 + RandomNum(0, 50);
+		set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+		circle(drawBitmap, sitePosX, sitePosY, circleRadius - 1, drawColor);
+	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ScenarioGUI::Draw(BITMAP *drawBitmap) const {
+		drawing_mode(DRAW_MODE_TRANS, nullptr, 0, 0);
+		DrawSitePoints(drawBitmap);
+		if (m_SelectedScene && m_ScenarioCollectionBoxes.at(ScenarioCollections::SceneInfoBox)->GetVisible()) { DrawLineToSitePoint(drawBitmap); }
+		drawing_mode(DRAW_MODE_SOLID, nullptr, 0, 0);
+
+		AllegroScreen drawScreen(drawBitmap);
+		m_ScenarioGUIController->Draw(&drawScreen);
+
+		// Draw scene preview after GUI
+		if (m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox)->GetVisible()) {
+			GUICollectionBox *playerSetupBox = m_ScenarioCollectionBoxes.at(ScenarioCollections::PlayerSetupBox);
+
+			// Draw the Player-Team matrix lines and disabled overlay effects.
+			int linePosY = 80;
+			for (int teamIndex = Activity::TeamOne; teamIndex < Activity::MaxTeamCount; ++teamIndex) {
+				if (m_SelectedActivity && (!m_SelectedActivity->TeamActive(teamIndex) || m_LockedCPUTeam == teamIndex)) {
+					// Apply a colored overlay on top of team rows that are not human-playable.
+					drawing_mode(DRAW_MODE_TRANS, nullptr, 0, 0);
+					int blendAmount = 230;
+					set_screen_blender(blendAmount, blendAmount, blendAmount, blendAmount);
+					rectfill(drawBitmap, playerSetupBox->GetXPos() + 110, playerSetupBox->GetYPos() + linePosY, playerSetupBox->GetXPos() + playerSetupBox->GetWidth() - 12, playerSetupBox->GetYPos() + linePosY + 25, c_GUIColorDarkBlue);
+					drawing_mode(DRAW_MODE_SOLID, nullptr, 0, 0);
+				}
+				// Cell border separator lines.
+				hline(drawBitmap, playerSetupBox->GetXPos() + 110, playerSetupBox->GetYPos() + linePosY, playerSetupBox->GetXPos() + playerSetupBox->GetWidth() - 12, c_GUIColorLightBlue);
+				linePosY += 25;
+			}
+
+			// Manually draw UI elements on top of colored rectangle.
```

Comment:
I don't think this comment adds much, it's super vague and the code does a better job of explaining that it's there to draw tech selector and ai skill level.
Could pull this out into a private submethod if you wanted though (same with the line/disabled overlay drawing above tbh).